src/Dscp/Educator/Secret/Real/Functions.hs

```diff
@@ -0,0 +1,104 @@
+-- | Functions to work with educator secret key storage.
+
+module Dscp.Educator.Secret.Real.Functions
+    ( linkStore
+    ) where
+
+import Data.Aeson (eitherDecode', encode)
+import qualified Data.ByteString.Lazy as LBS
+import System.Directory (doesFileExist)
+import System.FilePath ((</>))
+
+import Dscp.Crypto (PassPhrase, decrypt, encrypt, genSecretKey, runSecureRandom)
+import Dscp.Educator.Secret.Real.Error (EducatorSecretError (..), rewrapSecretIOError)
+import Dscp.Educator.Secret.Real.Types (EducatorSecret (..), EducatorSecretJson (..),
+                                        EducatorSecretParams (..), KeyfileContent)
+import Dscp.Resource.AppDir (AppDirectory (..))
+import Dscp.System (whenPosix)
+import Dscp.System (ensureModeIs, mode600)
+import Dscp.Util (leftToThrow)
+import Dscp.Util.Aeson (Versioned (..))
+
+---------------------------------------------------------------------
+-- Conversions
+---------------------------------------------------------------------
+
+toEducatorSecretJson :: PassPhrase -> EducatorSecret -> EducatorSecretJson
+toEducatorSecretJson pp EducatorSecret{..} =
+    let esjEncSecretKey = encrypt pp esSecretKey
+    in EducatorSecretJson{..}
+
+fromEducatorSecretJson :: MonadThrow m => PassPhrase -> EducatorSecretJson -> m EducatorSecret
+fromEducatorSecretJson pp EducatorSecretJson{..} = do
+    esSecretKey <- decrypt pp esjEncSecretKey
+        & leftToThrow SecretWrongPassPhraseError
+    return EducatorSecret{..}
+
+---------------------------------------------------------------------
+-- Storage operations
+---------------------------------------------------------------------
+
+-- | Where keyfile would lie.
+storePath :: EducatorSecretParams -> AppDirectory -> FilePath
+storePath EducatorSecretParams{..} (AppDirectory appDir) =
+    fromMaybe defPath espPath
+  where
+    defPath = appDir </> "secret.key"
```

Comment:
Hey, it should be `educator.key` (because in app directory there also would be `witness.key`, which is probably going to be different, and probably something else).
Also, there should be an option to set up a _completely_ custom absolute  path to the key, and from what I see here I conclude that it's only possible to choose app _directory_, but not the file's _basename_.

Reply:
Btw doesn't it mean that having `--keyfile-path` CLI option would be ambiguous?